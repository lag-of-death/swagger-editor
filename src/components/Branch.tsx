import * as React from "react";
import styled from "styled-components";
import { ILeafProps } from "./Leaf";

import { dotSize, sharedStyleForKey } from "./shared";

const Header = styled.div`
  ${sharedStyleForKey}
`;

const IssuesIndicatorsContainer = styled.div`
  height: ${dotSize};
`;

interface IBranchProps extends ILeafProps {
  render: () => React.FunctionComponent[];
}

const Branch = ({labelKey, render, renderIssuesIndicators}: IBranchProps) => {
  return (
    <li>
      <figure>
        <Header>
          <figcaption>
            {labelKey}
          </figcaption>

          <IssuesIndicatorsContainer>
            {renderIssuesIndicators()}
          </IssuesIndicatorsContainer>
        </Header>
        <ul>{render()}</ul>
      </figure>
    </li>
  );
};

export default Branch;
