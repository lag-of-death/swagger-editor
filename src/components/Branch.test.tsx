import { shallow } from "enzyme";
import * as React from "react";
import Branch from "./Branch";

describe("Branch", () => {
  test("rendering", () => {
    const component = shallow(
      <Branch render={() => []} labelKey="testLabelKey" renderIssuesIndicators={() => <div></div>}/>,
    );

    expect(component).toMatchSnapshot();
  });
});
