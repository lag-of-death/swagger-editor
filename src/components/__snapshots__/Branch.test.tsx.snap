// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branch rendering 1`] = `
<li>
  <figure>
    <styled.div>
      <figcaption>
        testLabelKey
      </figcaption>
      <styled.div>
        <div />
      </styled.div>
    </styled.div>
    <ul />
  </figure>
</li>
`;
